gwcs/wcs.py

```diff
@@ -502,13 +506,17 @@ def outside_footprint(self, world_arrays):
             ind = np.asarray((np.asarray(self.output_frame.axes_type) == axtyp))
 
             for idim, coord in enumerate(world_arrays):
-                coord = _tofloat(coord)
+                if not np.isscalar(coord):
+                    coord = _tofloat(coord.copy())
                 if np.asarray(ind).sum() > 1:
                     axis_range = footprint[:, idim]
                 else:
-                    axis_range = footprint
+                    axis_range = footprint[:, idim]
                 range = [axis_range.min(), axis_range.max()]
-                outside = (coord < range[0]) | (coord > range[1])
+                if axtyp == 'SPATIAL' and isinstance(self.output_frame, cf.CelestialFrame):
+                    range = np.mod(range, 180)
+                    coord = np.mod(coord, 180)
```

Comment:
Coordinates already _should be_ in the [0, 360] and [0, 180] (or [-90, 90]) range. Taking mod at 180 does not look like the correct approach. I think there may be cases when this could fail. For example, take CRVAL at 180 and an image that spans from ra=179.8 to 180.1.

Try adding some unit tests for this.

Also a good time to replace `range` with something else like `world_range`.